packages/drivers/odsp-driver/src/odspDelayLoadedDeltaStream.ts

```diff
@@ -220,9 +221,17 @@ export class OdspDelayLoadedDeltaStream {
 				this.currentConnection = connection;
 				return connection;
 			} catch (error) {
-				this.clearJoinSessionTimer();
-				this.cache.sessionJoinCache.remove(this.joinSessionKey);
-
+				// Remove join session information from cache only if it is an error is from socket event connect_document_error.
+				// Otherwise keep it in cache so that this session can be re-used after disconnection.
+				// Also keeping an undefined check here to account for any unknown code path that is unable to stamp the value as in that case also
+				// it is safer to clear join session cache and start over.
+				if (
+					(error as IAnyDriverError).eventName === "connect_document_error" ||
```

Comment:
You should check it like this:
```
if (
						typeof error === "object" &&
						error !== null &&
						(error as IAnyDriverError).eventName ===
							"connect_document_error"
					) { do stuff }

```

This will prevent unknown error which could end up closing the container. This way you would be sure that error is an object and that you can check props on it.

Reply:
For example, in case the error is just a string here, the above code would throw.